src/App.tsx

```diff
@@ -1,61 +1,123 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types';
+
+const getUser = (userId: number) => {
+  const foundedUser = usersFromServer.find(user => (
+    userId === user.id
+  ));
+
+  return foundedUser || null;
+};
+
+const todosWithUser: Todo[] = todosFromServer.map(todo => (
+  {
+    ...todo,
+    user: getUser(todo.userId),
+  }
+));
+
+const newId = (todos: Todo[]) => {
+  const maxId = (todos.map((todo) => (todo.id)));
+
+  return Math.max(...maxId) + 1;
+};
 
 export const App = () => {
+  const [user, setUser] = useState(0);
+  const [title, setTitle] = useState('');
+  const [todos, setTodos] = useState([...todosWithUser]);
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+
+  const handleTitleSelect = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setTitleError(false);
+  };
+
+  const handleUserSelect = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUser(+event.target.value);
+    setUserError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (!title) {
+      setTitleError(true);
+    }
+
+    if (!user) {
+      setUserError(true);
+    }
+
+    if (title !== '' && user !== 0) {
+      const newTodo: Todo = {
+        id: newId(todos),
+        userId: user,
+        title,
+        completed: false,
+        user: getUser(user),
+      };
+
+      setTodos([...todos, newTodo]);
+
+      setTitle('');
+      setUser(0);
+    }
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={(event) => (handleSubmit(event))}
```

Comment:
Why are you creating this anonymous function here ?

```suggestion
        onSubmit={handleSubmit}
```